collector-state/src/main/java/com/iluwatar/collectorstate/App.java

```diff
@@ -0,0 +1,39 @@
+package com.iluwatar.collectorstate;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * This is a brief structural introduction to Collector-State pattern using state machine
+ * with self-defined message and states.
+ */
+
+@Slf4j
+public final class App {
+
+  private App(){}
+
+  /**
+   * Program entry point.
+   *
+   * @param args command line args
+   */
+  public static void main(final String[] args) {
+    int expectedDigit = 8;
+    int minimumDigit = 2;
+    var collectorState = new DigitCollectorState(expectedDigit, minimumDigit);
+    var m = new StateMachine();

```

Comment:
It feels strange to have the state machine outside the DigitCollectorState. Maybe it should be baked in?